### ISSUE TITLE
One CacheManager with same name caches in builder

### ISSUE THREAD

**Issue by gibsong** [ehcache/ehcache3#2000]
When I create one CacheManager and try to create two caches with the same alias(one clustered and one not clustered) within this CacheManager only 1 cache is actually created!  This doesn't seem like the appropriate behavior in this case.  I create 2 caches but only get 1?  And more importantly I am unaware that I only have 1 cache.  It appears that the builder ends up using the last withCache(...) call and the previous ones are overwritten/ignored.

What is the expected behavior in this situation?  Should this throw an error?

Below is the configuration that exposed this issue:

    String cacheClusteredAlias = "cacheAliasSameName";
    String cacheUnclusteredAlias = "cacheAliasSameName";

    CacheConfiguration<Long, String> cacheConfigClustered = CacheConfigurationBuilder.newCacheConfigurationBuilder(Long, String, newResourcePoolsBuilder()
        .heap(10, MemoryUnit.MB)
        .offheap(50, MemoryUnit.MB)
        .with(ClusteredResourcePoolBuilder.clusteredDedicated("primary-server-resource", 100, MemoryUnit.MB)))
        .build();

    CacheConfiguration<Long, String> cacheConfigUnclustered = CacheConfigurationBuilder.newCacheConfigurationBuilder(Long, String, newResourcePoolsBuilder()
        .heap(10, MemoryUnit.MB)
        .offheap(100, MemoryUnit.MB))
        .build();

    cacheManager1 = CacheManagerBuilder.newCacheManagerBuilder()
        .with(ClusteringServiceConfigurationBuilder.cluster(URI.create(connectionURIStr + "/" + CACHE_MANAGER_NAME))
        .autoCreate()
        .defaultServerResource("primary-server-resource")
        .resourcePool("resource-pool-a", 128, MemoryUnit.MB, "secondary-server-resource")
        .resourcePool("resource-pool-b", 128, MemoryUnit.MB))
        .withCache(cacheClusteredAlias, cacheConfigClustered)
        .withCache(cacheUnclusteredAlias, cacheConfigUnclustered)
        .using(new DefaultManagementRegistryConfiguration()
        .setCacheManagerAlias(CACHE_MANAGER_NAME)).build(true);

**Comment by GaryWKeim** [comment]
You describe using the same alias (why would you do this?) but your code snippet implies different ones. Clearly, if you used the same alias that wouldn't work because it's probably a map key somewhere.

**Comment by henri-tremblay** [comment]
@GaryWKeim I think he just wanted to file issue #2000. Joke aside, it should not work indeed. I've simplified the test case to this

```java
  @Test
  public void test() {
    String cacheAlias = "cacheAliasSameName";

    CacheConfiguration<Long, String> cacheConfig = CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class, newResourcePoolsBuilder()
      .heap(10, MemoryUnit.MB)
      .offheap(50, MemoryUnit.MB))
      .build();

    CacheManager cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
      .withCache(cacheAlias, cacheConfig)
      .withCache(cacheAlias, cacheConfig)
      .build(true);
  }
```

It should indeed return an error. Funnily, `cacheManager.createCache(cacheAlias, cacheConfig);` does return an error.

**Comment by GaryWKeim** [comment]
Agreed that it should error out or provide some feedback.

**Comment by gibsong** [comment]
Great, glad to hear some consensus on this issue.  @henri-tremblay  Thanks for doing a little digging and verifying it is in fact a bug.

**Comment by ljacomet** [comment]
Needs to be daggy fixed to 3.2, 3.3 and master
